plugins/woocommerce-blocks/bin/gen-block-list-doc.js

```diff
@@ -0,0 +1,206 @@
+
+/**
+ * Regular expression using tokens for matching in doc file.
+ * Note: `.` does not match new lines, so [^] is used.
+ *
+ * @type {RegExp}
+ */
+const TOKEN_PATTERN = new RegExp( START_TOKEN + '[^]*' + END_TOKEN );
+
+/**
+ * Returns list of keys, filtering out any experimental
+ * and wrapping keys with ~~ to strikeout false values.
+ *
+ * @type {Object} obj
+ * @return {string[]} Array of truthy keys
+ */
+function getTruthyKeys( obj ) {
+	return Object.keys( obj )
+		.filter( ( key ) => ! key.startsWith( '__exp' ) )
+		.map( ( key ) => ( obj[ key ] ? key : `~~${ key }~~` ) );
+}
+
+/**
+ * Process list of object that may contain inner keys.
+ * For example: spacing( margin, padding ).
+ *
+ * @param {Object} obj
+ * @return {string[]} Array of keys (inner keys)
+ */
+function processObjWithInnerKeys( obj ) {
+	const rtn = [];
+
+	const kvs = getTruthyKeys( obj );
+
+	kvs.forEach( ( key ) => {
+		if ( Array.isArray( obj[ key ] ) ) {
+			rtn.push( `${ key } (${ obj[ key ].sort().join( ', ' ) })` );
+		} else if ( typeof obj[ key ] === 'object' ) {
+			const innerKeys = getTruthyKeys( obj[ key ] );
+			rtn.push( `${ key } (${ innerKeys.sort().join( ', ' ) })` );
+		} else {
+			rtn.push( key );
+		}
+	} );
+	return rtn;
+}
+
+/**
+ * Augment supports with additional default props.
+ *
+ * The color support if specified defaults background and text, if
+ * not disabled. So adding { color: 'link' } support also brings along
+ * background and text.
+ *
+ * @param {Object} supports - keys supported by blokc
+ * @return {Object} supports augmented with defaults
+ */
+function augmentSupports( supports ) {
+	if ( supports && 'color' in supports ) {
+		// If backgroud or text is not specified (true or false)
+		// then add it as true.a
+		if (
+			typeof supports.color === 'object' &&
+			! ( 'background' in supports.color )
+		) {
+			supports.color.background = true;
+		}
+		if (
+			typeof supports.color === 'object' &&
+			! ( 'text' in supports.color )
+		) {
+			supports.color.text = true;
+		}
+	}
+	return supports;
+}
+
+/**
+ * Reads block.json file and returns markdown formatted entry.
+ *
+ * @param {string} filename
+ *
+ * @return {string} markdown
+ */
+function readBlockJSON( filename ) {
+	const blockjson = require( filename );
+	let supportsList = [];
+	let attributes = [];
+
+	if ( typeof blockjson.name === 'undefined' ) {
+		return ``;
+	}
+
+	if ( typeof blockjson.supports !== 'undefined' ) {
+		const supportsAugmented = augmentSupports( blockjson.supports );
+
+		if ( supportsAugmented ) {
+			supportsList = processObjWithInnerKeys( supportsAugmented );
+		}
+	}
+
+	if ( typeof blockjson.attributes !== 'undefined' ) {
+		attributes = getTruthyKeys( blockjson.attributes );
+	}
+
+	return `
+## ${ blockjson.title } - ${ blockjson.name }
+
+${ blockjson.description || '' }
+
+-	**Name:** ${ blockjson.name }
+-	**Category:** ${ blockjson.category || '' }
+-   **Ancestor:** ${ blockjson.ancestor || '' }
```

Comment:
I just realised some blocks specify `parent` metadata which makes them available only when directly nested within specific block. Worth adding that field as well.

```suggestion
-   **Ancestor:** ${ blockjson.ancestor || '' }
-   **Parent:** ${ blockjson.parent || '' }
```